File: Golang/PalindromeNumber/PalindromeNumber_test.go
package PalindromeNumber

import "testing"

func TestIsPalindrome(t *testing.T) {

	if isPalindrome(-1) == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isPalindrome(1) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isPalindrome(121) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isPalindrome(1001) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isPalindrome(1000021) == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/PalindromeNumberTest.py
from unittest import TestCase

from PalindromeNumber import PalindromeNumber


class TestPalindromeNumber(TestCase):
    def test_isPalindrome(self):
        pn = PalindromeNumber()

        self.assertFalse(pn.isPalindrome(-1))

        self.assertTrue(pn.isPalindrome(1))

        self.assertTrue(pn.isPalindrome(121))

        self.assertTrue(pn.isPalindrome(1001))

        self.assertFalse(pn.isPalindrome(1000021))
